tagstory/src/main/java/no/tagstory/utils/Database.java

```diff
@@ -147,4 +194,63 @@ public boolean insertStatistic(String storyId, Date startTime, long duration, in
 		values.put(STATISTICS_DISTANCE, distance);
 		return db.insert(STATISTICS_TABLE_NAME, null, values) != -1;
 	}
+
+	public boolean insertSaveAndQuit(String statisticId, String historyRootId) {
+		ContentValues mValues = new ContentValues(2);
+		mValues.put(SAVE_QUIT_STATISTIC_ID, statisticId);
+		mValues.put(SAVE_QUIT_HISTORY_ROOT_ID, historyRootId);
+		return db.insert(SAVE_QUIT_TABLE_NAME, null, mValues) != -1;
+	}
+
+	public Cursor getSaveAndQuit(String historyRootId) {
+		return db.query(
+				SAVE_QUIT_TABLE_NAME, new String[] {SAVE_QUIT_ID},
+				SAVE_QUIT_HISTORY_ROOT_ID + "=?", new String[]{historyRootId},
+				null, null, null);
+	}
+
+	public boolean deleteSaveAndQuit(String sqId) {
+		int result = db.delete(SAVE_QUIT_TABLE_NAME, SAVE_QUIT_HISTORY_ROOT_ID
+				+ "=?", new String[]{sqId});
+		return result > 0;
+	}
+
+	public boolean insertLocation(String sqId, double latitude, double longitude) {
+		ContentValues mValues = new ContentValues(3);
+		mValues.put(LOCATIONS_SAVE_QUIT_ID, sqId);
+		mValues.put(LOCATIONS_LATITUDE, latitude);
+		mValues.put(LOCATIONS_LONGITUDE, longitude);
+		return db.insert(LOCATIONS_TABLE_NAME, null, mValues) != -1;
+	}
+
+	public Cursor getLocation() {
+		return db.query(LOCATIONS_TABLE_NAME, null, null, null, null, null,
+				null);
+	}
+
+	public boolean deleteLocation(String locationId) {
+		return db.delete(LOCATIONS_TABLE_NAME, LOCATIONS_ID + "=?",
+				new String[]{locationId}) > 0;
+	}
+
+	public boolean insertHistory(String storyId, String historyPrevious,
```

Comment:
Kan du ikke sende inn hele noden istedenfor? Og så kalle metoden for `insertHistoryNode`.